scripts/run.py

```diff
@@ -400,40 +400,29 @@ def start(self):
     def scrape_port_from_stdout(self):
         """
         Look at the stdout log and figure out which port the JVM chose.
-        Write this to self.port.
-        This call is blocking.
-        Exit if this fails.
 
-        :return none
-        """
-        retries = 30
-        while retries > 0:
-            if self.terminated: return
-            f = open(self.output_file_name, "r")
-            s = f.readline()
-            while len(s) > 0:
-                if self.terminated: return
-                match_groups = re.search(r"Listening for HTTP and REST traffic on http.*://(\S+):(\d+)", s)
-                if match_groups is not None:
-                    port = match_groups.group(2)
-                    if port is not None:
-                        self.port = port
-                        f.close()
-                        print("H2O Cloud {} Node {} started with output file {}".format(self.cloud_num,
-                                                                                        self.node_num,
-                                                                                        self.output_file_name))
+        If successful, port number is stored in self.port; otherwise the
+        program is terminated. This call is blocking, and will wait for
+        up to 30s for the server to start up.
+        """
+        regex = re.compile(r"Open H2O Flow in your web browser: https?://([^:]+):(\d+)")
+        retries_left = 30
+        while retries_left and not self.terminated:
```

Comment:
Is this pattern useful somewhere else? as well? for example job progress? what if job progress `GET` call fails - do we try to retry?